pkg/controllers/user/controllers.go

```diff
@@ -72,7 +72,7 @@ func Register(ctx context.Context, cluster *config.UserContext, clusterRec *mana
 	istio.Register(ctx, cluster)
 	certsexpiration.Register(ctx, cluster)
 
-	if clusterRec.Spec.LocalClusterAuthEndpoint.Enabled {
+	if clusterRec.Spec.LocalClusterAuthEndpoint.Enabled && managementv3.ClusterConditionReady.IsTrue(clusterRec) {
```

Comment:
What happens if cluster is not ready? Will it re-register when it becomes ready? 

Would it be better to create CRDs with factory.CreateCRDs which returns errors rather than factory.BatchCreateCRDs which panics? Or just return an error if the cluster is not ready?

Reply:
@erikwilson Actually, it doesn't seem this will work. At start up, `managementv3.ClusterConditionReady.IsTrue(clusterRec)` is always false, regardless of the cluster actual state. 

CC: @alena1108